LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -672,10 +672,11 @@ extension DoseStore {
 
      - parameter events: An array of new pump events. Pump events should have end times reflective of when delivery is actually expected to be finished, as doses that end prior to a reservoir reading are ignored when reservoir data is being used.
      - parameter lastReconciliation: The date that pump events were most recently reconciled against recorded pump history. Pump events are assumed to be reflective of delivery up until this point in time. If reservoir values are recorded after this time, they may be used to supplement event based delivery.
+     - parameter replacePendingEvents: If true, any existing pending events will be removed.
      - parameter completion: A closure called after the events are saved. The closure takes a single argument:
      - parameter error: An error object explaining why the events could not be saved.
      */
-    public func addPumpEvents(_ events: [NewPumpEvent], lastReconciliation: Date?, completion: @escaping (_ error: DoseStoreError?) -> Void) {
+    public func addPumpEvents(_ events: [NewPumpEvent], lastReconciliation: Date?, replacePendingEvents: Bool = true, completion: @escaping (_ error: DoseStoreError?) -> Void) {
```

Comment:
should this default to `false`? Some pump managers keep pump events (e.g., temp basal) as mutable until they are known to finish, even if another event (e.g., bolus) occurs in the meantime. Maybe I'm overlooking something?

Ah... I see in the comment that events that are still valid should be reported again. Should that be added to the comments on line 675? Maybe this is all that is needed. But it is important to know that pending pump events will get removed by default unless they are repeatedly reported.